### PR TITLE
 [CDAP-13012] Support customizing the text input format in file source

### PR DESCRIPTION
**Pull request by anew**

### CODE REVIEW

File: core-plugins/docs/File-batchsource.md [hunk 1]
```diff
@@ -51,6 +51,11 @@ Cannot be used if pathField is set. (Macro-enabled)
 
 **format:** Format of the file. Must be "text", "avro" or "parquet". Defaults to "text".
 
+**delegateTextInputFormat:** The fully qualified class name of a Hadoop input format to be used for reading
+the input file(s) if the format is 'text'. This is useful for text file formats that require special handling
+not available from the default TextInputFormat. The class must extend ``InputFormat<LongWritable, Text>`` and use
+splits that extend ``FileSplit``. It must be available to the program class loader at run time.
```
Comment (sreevatsanraman) [inline core-plugins/docs/File-batchsource.md:57]: ```It must be available to the program class loader at run time.``` - Maybe we can rephrase it to let the users know what the exact steps are in making the class available.

**Comment by anew** [comment]
closing this for now, as we do not have agreement on this feature.
